### PR TITLE
Enforce required permissions when adding/editing known Schemes

### PR DESCRIPTION
**Pull request by tibetsprague**
Also move all scheme utilities into a new lib/schemeUtils.ts file

### CODE REVIEW

File: src/lib/schemeUtils.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+import { getArc } from "../arc";
+
+export enum SchemePermissions {
+  None = 0,
+  IsRegistered = 1, // Always added by default in the controller
+  CanRegisterSchemes = 2,
+  CanAddRemoveGlobalConstraints = 4,
+  CanUpgradeController = 8,
+  CanCallDelegateCall = 0x10,
```
Comment (dkent600) [inline src/lib/schemeUtils.ts]: There is also a metadata permission that I'm querying Oren about.

Comment (tibetsprague) [reply src/lib/schemeUtils.ts]: sounds like this is all good for now

Comment (dkent600) [reply src/lib/schemeUtils.ts]: I haven't seen a response from Oren about the metadata permission (that it overlaps the delegatecall permission).

Comment (tibetsprague) [reply src/lib/schemeUtils.ts]: the responded and said they are the same

Comment (dkent600) [reply src/lib/schemeUtils.ts]: Ah, got it, looks good

**Pull request merged**